driver/controller.go

```diff
@@ -963,3 +973,46 @@ func validateCapabilities(caps []*csi.VolumeCapability) bool {
 
 	return supported
 }
+
+func (d *Driver) tagVolume(ctx context.Context, vol *godo.Volume) error {
+	for _, tag := range vol.Tags {
+		if tag == d.doVolTag {
+			return nil
+		}
+	}
+
+	tagReq := &godo.TagResourcesRequest{
+		Resources: []godo.Resource{
+			godo.Resource{
+				ID:   vol.ID,
+				Type: godo.VolumeResourceType,
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, doAPITimeout)
+	resp, err := d.tags.TagResources(ctx, d.doVolTag, tagReq)
+	cancel()
```

Comment:
I believe you should call this in a `defer`, otherwise the context will be cancelled immediately, which is also passed down below.

Reply:
Good catch on the context reuse.  I was meaning to use the parent context for each API call rather than chaining them like that.  Tests would have missed that since we faked the TagService.

I struggled on the right way to deal with context here.  From my perspective, as long as we're making progress we should reset the timeout for each call.  So I didn't want to use a single context/cancel. 

I had originally opted to leave the `cancel()` bare rather than in a defer because I thought deferring a cancel() and then overriding that cancel function-ref with a new cancel for the next call was confusing.  `defer cancel()` dereferences the function-ref at the line where the defer is; not the end of the function.  Using the defer does mean those first two timeouts might fire (with no effect) after their associated API call completes but before the defer cancels them.  The only harm is a fairly trivial amount of inefficiency in exchange for a lot of clarity.